pkg/controller/jobs/statefulset/statefulset_reconciler.go

```diff
@@ -144,15 +159,52 @@ func (r *Reconciler) Update(e event.UpdateEvent) bool {
 	return r.handle(e.ObjectNew)
 }
 
-func (r *Reconciler) Delete(event.DeleteEvent) bool {
-	return false
+func (r *Reconciler) Delete(e event.DeleteEvent) bool {
+	return r.handle(e.Object)
 }
 
 func (r *Reconciler) handle(obj client.Object) bool {
 	sts, isSts := obj.(*appsv1.StatefulSet)
 	if !isSts {
-		return false
+		return true
 	}
 	// Handle only statefulset managed by kueue.
 	return jobframework.IsManagedByKueue(sts)
 }
+
+var _ handler.EventHandler = (*podHandler)(nil)
+
+type podHandler struct{}
+
+func (h *podHandler) Generic(context.Context, event.GenericEvent, workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+}
+
+func (h *podHandler) Create(_ context.Context, e event.CreateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	// To watch for pod ADDED events.
+	h.handle(e.Object, q)
```

Comment:
Couldn't we directly implement logic here instead of separated `handle` function?

Reply:
Done